src/hooks/useAnchoredPosition.ts

```diff
@@ -27,17 +27,36 @@ export function useAnchoredPosition(
   const floatingElementRef = useProvidedRefOrCreate(settings?.floatingElementRef)
   const anchorElementRef = useProvidedRefOrCreate(settings?.anchorElementRef)
   const [position, setPosition] = React.useState<AnchorPosition | undefined>(undefined)
-  React.useLayoutEffect(() => {
-    if (floatingElementRef.current instanceof Element && anchorElementRef.current instanceof Element) {
-      setPosition(getAnchoredPosition(floatingElementRef.current, anchorElementRef.current, settings))
-    } else {
-      setPosition(undefined)
-    }
+
+  const updatePosition = React.useCallback(
+    () => {
+      if (floatingElementRef.current instanceof Element && anchorElementRef.current instanceof Element) {
+        setPosition(getAnchoredPosition(floatingElementRef.current, anchorElementRef.current, settings))
+      } else {
+        setPosition(undefined)
+      }
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [floatingElementRef, anchorElementRef, ...dependencies])
+    [floatingElementRef, anchorElementRef, ...dependencies]
+  )
+
+  React.useLayoutEffect(updatePosition, [updatePosition])
+
+  useResizeObserver(updatePosition)
+
   return {
     floatingElementRef,
     anchorElementRef,
     position
   }
 }
+
+const useResizeObserver = (callback: () => void) => {
```

Comment:
This is a scrappy resize observer hook. There are small npm packages that offer similar hooks with additional features like:

- keeping a singleton resize observer
- debouncing/throttling
- polyfilling browsers that don't support ResizeObservers